rca/people/admin_forms.py

```diff
@@ -8,6 +8,19 @@ class StudentPageAdminForm(WagtailAdminPageForm):
     """Validation for adding StudentPage
     """
 
+    def __init__(self, *args, **kwargs):
+        print(args)
+        print(kwargs)
+        super(StudentPageAdminForm, self).__init__(*args, **kwargs)
+        # TODO work in request.user.group IS/HAS Student and lock fields
+        readonly_fields = [
+            "first_name",
+            "last_name",
+        ]
```

Comment:
This is blocking the last AC in the task. I need to work in a way of getting the user requesting the form here
so I can conditionally lock the fields depending on the users role.

Reply:
Some of the answers on [this Stackoverflow post](https://stackoverflow.com/questions/2864955/django-how-to-get-current-user-in-admin-forms) might help you here

Reply:
Marking this as resolved, we are actually tackling this in the next ticket, I'm getting ahead of myself